secret/secret_test.go

```diff
@@ -236,28 +281,35 @@ func TestGetOrCreateWithValidSecrets(t *testing.T) {
 		logFatal = mockLog.Fatal
 
 		getEnv = func(string) string {
-			return ""
+			// RELEASE_VERSION
+			return "3"
 		}
-		sMissing.On("Get", SECRET_UPDATE_NAME, metav1.GetOptions{})
+
+		sMissing.On("Get", SECRET_UPDATE_NAME+"-3", metav1.GetOptions{})
+		sMissing.On("Delete", SECRET_NAME+"-1", &metav1.DeleteOptions{})
+		sMissing.On("Get", SECRET_NAME+"-2", metav1.GetOptions{})
 		sMissing.On("Get", SECRET_NAME, metav1.GetOptions{})
```

Comment:
Why is this still needed; I would think `Get` would never be called with an unversioned `secret` name because fetching `secret-2` succeeded.

Same thing may be true for other tests too; please double-check.

Reply:
Checking... This might be the test where I try to force the fallback to unversioned `secret`, and `sMissing`s `Get` is modified so that `secret-2` fails. If so, I will try to make that more clear

Reply:
Addressed via commit 65e5b28442625a1212030de22f7007b3bbed17d4
This is the test where both `secret` and `secret-N` are missing.
Tweaked test description and added comments cross-referencing the test with the relevant `Get` in the mocks.